src/libasr/bwriter.h

```diff
@@ -97,6 +157,21 @@ class BinaryWriter
         write_int64(*ip);
     }
 
+    void write_uintptr(uintptr_t i) {
+        s.append(uintptr_to_string(i));
+    }
+
+    void write_void(void *p, int64_t m_n_data, ASR::ttype_t* m_type) {
+        int kind = ASRUtils::extract_kind_from_ttype_t(m_type);
+        int n_data = m_n_data / kind;
+
+        for (int64_t i = 0; i < n_data; i++) {
+            void* p_i = (void*)((uintptr_t)p + i*kind);
+            uint64_t* p_i_64 = (uint64_t*)p_i;
+            write_int64(*p_i_64);
+        }
+    }
```

Comment:
I think you are encoding 1 byte into 8 bytes (wasting 7 bytes?). Why not just this:
```suggestion
        for (int64_t i = 0; i < n_data; i++) {
            uint8_t* p_i8 = (uint8_t*)p;
            write_int8(p_i8[i]);
        }
    }
```

Reply:
NIce, thanks for pointing this :))